### PR TITLE
Introduce methods for setting up static ipv6

### PR DESCRIPTION
**Pull request by isimluk**
# What
 - Introduce `address6=` method for setting static IPv6 address in /etc/sysconfig/network-scripts/ifcfg
 - introduce `gateway6=` method
 - introduce `apply_static6` method

### CODE REVIEW

File: lib/linux_admin/network_interface/network_interface_rh.rb [hunk 1]
```diff
@@ -104,6 +125,23 @@ def apply_static(ip, mask, gw, dns, search = nil)
...
+    # @param gw [String] gateway address
+    # @param dns [Array<String>] list of dns servers
+    # @param search [Array<String>] list of search domains
+    # @return [Boolean] true on success, false otherwise
+    # @raise ArgumentError if an IP is not formatted properly or interface does not start
+    def apply_static6(ip, prefix, gw, dns, search = nil)
+      self.address6     = "#{ip}/#{prefix}"
+      self.gateway6     = gw
+      self.dns          = dns
+      self.search_order = search
```
Comment (carbonin) [inline lib/linux_admin/network_interface/network_interface_rh.rb]: Should we have the same check here as we do in `#apply_static`?

Comment (isimluk) [inline lib/linux_admin/network_interface/network_interface_rh.rb]: Nice catch, fixed. ✏️

**Comment by isimluk** [comment]
> Should we remove some of the IPv6 specific fields in enable_dhcp?

I don't know for sure. Nothing should break if we don't.

From what I know today, it seems that dhcpv4 and static IPv6 could work side by side. I plan to dive deeper in DHCPv6 once the static stuff is merged.

It is not clear to me whether `BOOTPROTO='static'` applies to IPv6 or not. Needs some experiments.

**Comment by carbonin** [comment]
Just kicking to make sure hakiri is fixed.

**Comment by isimluk** [comment]
Rebased to get harakiri green.

File: spec/network_interface/network_interface_rh_spec.rb [hunk 2]
```diff
@@ -100,6 +100,21 @@ def result(output, exit_status)
...
+  describe '#address6=' do
+    it 'sets the ipv6 address' do
+      address = 'fe80::1/64'
+      dhcp_interface.address6 = address
+      conf = dhcp_interface.interface_config
+      expect(conf['IPV6ADDR']).to eq(address)
+      expect(conf['BOOTPROTO']).to eq('static')
+      expect(conf['IPV6INIT']).to eq('yes')
+      expect(conf['DHCPV6C']).to eq('no')
+    end
```
Comment (carbonin) [inline spec/network_interface/network_interface_rh_spec.rb]: Just missing a space :smile:

**Comment by isimluk** [comment]
Thanks!! I have brought in the lost new line.

**Comment by isimluk** [comment]
Thanks!

**Pull request merged**